### PR TITLE
some tests and readme

### PR DESCRIPTION
**Pull request by sarimabbas**
added tests to test_utils_lib, begun test_gripper, created read me, modified CMakeLists for tests

### CODE REVIEW

**Comment by alecive** [comment]
@sarimabbas nice pic! In general, it is a good practice to do smaller commits that are self-contained (eg one for the readme, one for the unit tests), and to push often, so that I don't have to review everything altogether.

Btw for now it's okay, I'll wait for travis and codacy and then I'll do my review.

**Comment by sarimabbas** [comment]
Oh ok, will do next time!

File: baxter_collaboration_lib/CMakeLists.txt
Comment (alecive) [inline baxter_collaboration_lib/CMakeLists.txt:149]: Add comment as you did with `Robot interface tests` and "Robot util test"

File: baxter_collaboration_lib/test/README.md
Comment (alecive) [inline baxter_collaboration_lib/test/README.md]: Remove one level of heading (also below)

File: baxter_collaboration_lib/test/README.md
Comment (alecive) [inline baxter_collaboration_lib/test/README.md]: `robotInterface` tests still need to be done.

File: baxter_collaboration_lib/test/test_gripper.cpp
Comment (alecive) [inline baxter_collaboration_lib/test/test_gripper.cpp]: This is not necessarily your fault, but prefer `EXPECT_FALSE` to `EXPECT_EQ(false, `. If that appeared in `test_robot_interface`, please change that as well.

File: baxter_collaboration_lib/test/test_gripper.cpp
Comment (alecive) [inline baxter_collaboration_lib/test/test_gripper.cpp]: Simliarly, use `EXPECT_TRUE` here.

File: baxter_collaboration_lib/test/test_gripper.cpp
Comment (alecive) [inline baxter_collaboration_lib/test/test_gripper.cpp]: Delete these comments. These should be tested in the simulator test.

File: baxter_collaboration_lib/test/test_utils_lib.cpp
Comment (alecive) [inline baxter_collaboration_lib/test/test_utils_lib.cpp]: Good job here! But try also to think about edge cases, eg negative numbers, zeros, things that might break the code.

**Review by alecive — CHANGES_REQUESTED** [review]

**Comment by sarimabbas** [comment]
Have reviewed everything except the edge cases for test_utils_lib, will think about them in detail today!

File: baxter_collaboration_lib/test/test_utils_lib.cpp
Comment (alecive) [inline baxter_collaboration_lib/test/test_utils_lib.cpp]: Remove this comment, it's not useful :)

File: baxter_collaboration_lib/test/README.md
Comment (alecive) [inline baxter_collaboration_lib/test/README.md]: I would probably move the checklist out of the `README`. Open an issue and place it there instead.

**Review by alecive — CHANGES_REQUESTED** [review]

File: baxter_collaboration_lib/include/robot_utils/utils.h
Comment (alecive) [inline baxter_collaboration_lib/include/robot_utils/utils.h]: What is this function for?

File: baxter_collaboration_lib/test/README.md
Comment (alecive) [inline baxter_collaboration_lib/test/README.md]: Why`moveit`?

**Review by alecive — CHANGES_REQUESTED** [review]

File: baxter_collaboration_lib/include/robot_utils/utils.h
Comment (alecive) [reply baxter_collaboration_lib/include/robot_utils/utils.h]: I'm not saying that you should not have done it, but please justify :)

Comment (sarimabbas) [reply baxter_collaboration_lib/include/robot_utils/utils.h]: Sure. I wanted to preserve the code that was already present that used stringstream. But the str() method with stringstream would not convert something like -0.0 to 0 (instead it would give -0). So for just the toString(double) situation, the fix was easy enough, but for the toString(doubleVector) and toString(Pose) situations, there was the possibility of multiple -0s being present in the final string. So I thought I'd create a new function in utils that I could use for both situations.

Within the toString functions, a vector like [-0.0, 1, 2, 3, -0] now becomes [0, 1, 2, 3, 0]. Same for a Pose. I tried to think of potential dangers of such a function, but in my use, I have only targeted "-0," and "-0}" substrings for replacement.

File: baxter_collaboration_lib/test/README.md
Comment (sarimabbas) [reply baxter_collaboration_lib/test/README.md]: When Andrew and I were trying to get him set up, he was unable to build baxter_collaboration without moveit. On the other hand, his trac_ik package was not building, so perhaps if his trac_ik builds, moveit may not be required. So I can defer this till tomorrow?

Comment (alecive) [reply baxter_collaboration_lib/test/README.md]: Yes. Moveit is not needed in our software, let's take a look at that together.

File: baxter_collaboration_lib/include/robot_utils/utils.h
Comment (alecive) [reply baxter_collaboration_lib/include/robot_utils/utils.h]: What if you change `toString(vector)` into a for loop that cycles through the elements and does individual `toString(double)`? Overall, this is a very minor function, whose behavior is not critical since it is only used for printouts. I would try to move on with this as fast as possible and start testing more important things.

Comment (sarimabbas) [reply baxter_collaboration_lib/include/robot_utils/utils.h]: You're right, this was a cleaner approach which I have implemented :)

**Comment by alecive** [comment]
Ok feel free to merge, delete the branch and work on a new thing in a new branch.

**Comment by sarimabbas** [comment]
Great! How would i do the merge? The button is greyed out on my end.

**Review by alecive — APPROVED** [review]

**Comment by alecive** [comment]
Ops sorry now it should be working.

**Comment by sarimabbas** [comment]
screenshot from 2017-05-23 12 57 40

Sorry but even after your approval it is greyed out

**Pull request merged**
